src/hooks/usePopper.ts

```diff
@@ -0,0 +1,136 @@
+import { useRef, useEffect, RefObject, useState, useMemo, useLayoutEffect } from 'react'
+import { Instance, Options, State, Placement } from '@popperjs/core'
+
+export const TOOLTIP_OFFSET = 8 // px
+
+const defaultConfig: Partial<Options> & Pick<Options, 'modifiers'> = {
+  // default tooltip placement
+  placement: 'top',
+  modifiers: [
+    {
+      // move slightly to make space for the arrow element
+      name: 'offset',
+      options: {
+        offset: [0, 8],
+      },
+    },
+  ],
+}
+
+const createConfig = (
+  config: Partial<Options> | undefined | null,
+  setState: (state: State) => void,
+): Partial<Options> => {
+  const finalConfig = {
+    ...defaultConfig,
+    ...config,
+  }
+
+  // applyStyles delegates styling to react
+  finalConfig.modifiers.push({
+    name: 'applyStyles',
+    fn: ({ state }) => {
+      // popper.state is mutable, need a new instance
+      setState({ ...state })
+    },
+  })
+
+  return finalConfig
+}
+
+interface PopperHookResult<T extends HTMLElement, U extends HTMLElement = HTMLDivElement> {
```

Comment:
Do we have a convention  for this interfaces? 
Personally I think it's fine to call it just Result, specially if you don't export it

Reply:
Renamed